src/models/treemodel.h

```diff
@@ -281,6 +278,46 @@ class CostTreeModel : public TreeModel<decltype(Results::root), ModelImpl>
     Results m_results;
 };
 
+template<typename Results, typename ModelImpl>
```

Comment:
this should be in a separate header and not be a template. instead pass the respective SymbolRole and use that below